### PR TITLE
perf: improve php_server directive

### PR DESCRIPTION
**Pull request by dunglas**
Prevents some I/O accesses when disabling the file server and when disabling index files in subdirectory support.

### CODE REVIEW

File: caddy/caddy.go [hunk 1]
```diff
@@ -546,31 +546,51 @@ func parsePhpServer(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error)
...
-			MatcherSetsRaw: []caddy.ModuleMap{redirMatcherSet},
-			HandlersRaw:    []json.RawMessage{caddyconfig.JSONModuleObject(redirHandler, "handler", "static_response", nil)},
-		}
+		var disableDirRedir bool
 
 		// if tryFiles wasn't overridden, use a reasonable default
 		if len(tryFiles) == 0 {
-			tryFiles = []string{"{http.request.uri.path}", "{http.request.uri.path}/" + indexFile, indexFile}
+			if disableFsrv {
+				tryFiles = []string{"{http.request.uri.path}/" + indexFile, indexFile}
```
Comment (dunglas) [inline caddy/caddy.go:554]: This change beaks the tests.

Comment (dunglas) [reply caddy/caddy.go:554]: Actually, the new behavior is better and more correct than the previous one. I updated the test accordingly.

**Comment by AlliBalliBaba** [comment]
I wonder if it would also make sense to cache the location of existing index files (unless that's something you're already doing here). The only disadvantage would be that when the index file is deleted, we'd get a PHP 505 file not found instead of a 404.

**Comment by dunglas** [comment]
I think it's a good idea. We'll have to introduce a LRU cache in Caddy to do that. It may also be needed for another optimization I have in mind: caching if files (assets) exist or not for some time (configurable).

That being said, I'm currently optimizing for the framework case (Laravel/Symfony), and I think that for this specific (but very common) case I'll be able to get 0 to disk access with https://github.com/caddyserver/caddy/pull/6699 and some extra tricks to come.

**Pull request merged**